compiler/rustc_feature/src/active.rs

```diff
@@ -638,6 +638,10 @@ declare_features! (
     /// Allows macro attributes to observe output of `#[derive]`.
     (active, macro_attributes_in_derive_output, "1.51.0", Some(81119), None),
 
+    /// Allows the use of top-level or-patterns in `let` if there is no type annotation.
+    /// Ex: `let (A | B) : Ty = foo;` or `let A | B = foo;` but not `let A | B : Ty;`.
+    (active, or_patterns_with_ty, "1.52.0", Some(54883), None),
```

Comment:
This is such an arbitrary and special-purpose feature gate.
I think it's better to just make this a hard error and suggest wrapping into parentheses.

`let P1 | P2 = foo` is a rare construction, so adding parentheses would require less symbols than writing the feature even on nightly.